RoboFile.php

```diff
@@ -531,6 +573,46 @@ public function tryTmpDir()
 
         return $result;
     }
+
+    public function tryProgress()
+    {
+        $processList = range(1, 10);
+        return $this->builder()
+            ->taskForEach($processList)
+                ->iterationMessage('Processing {value}')
+                ->call(
+                    function ($value) {
+                        // TaskForEach::call should only be used to do
+                        // non-Robo operations. To use Robo tasks in an
+                        // iterator, @see TaskForEach::withBuilder.
+                        usleep(500000); // sleep for half a second
+                    }
+                );
+    }
+
+    public function tryIter()
+    {
+        $delayUntilProgressStart = 2;
+        TaskBuilder::setDefaultProgressBarAutoDisplayInterval($delayUntilProgressStart);
+        $this->say("Progress bar will display after $delayUntilProgressStart seconds of activity.");
+
+        $workdir = 'build/iter-example';
+        $this->say("Creating sample direcories in $workdir.");
+
+        $processList = ['cats', 'dogs', 'sheep', 'fish', 'horses', 'cows'];
+        return $this->builder()
+            ->taskFilesystemStack()
+                ->mkdir($workdir)
+            ->taskCleanDir($workdir)
+            ->taskForEach($processList)
+                ->withBuilder(
+                    function ($builder, $key, $value) use ($workdir) {
+                        return $builder
+                            ->taskFilesystemStack()
+                                ->mkdir("$workdir/$value");
```

Comment:
Yeah, the strong inversion of control here is very reminiscent of node.js.  Overall, I prefer an Iterator object over a function that takes a callback object; however, in this instance, the fact that we are managing progress and failures in the collection necessitates this design.  This is in fact also similar to the design decisions that drove node.js, although the reasons are different.